include/bluetooth/audio/vocs.h

```diff
@@ -0,0 +1,302 @@
+
+	/** Initial volume offset (-255 to 255) */
+	int16_t offset;
+
+	/** Initial audio output description */
+	char *output_desc;
+
+	/** Boolean to set whether the description is writable by clients */
+	bool desc_writable;
+};
+
+/** @brief Structure for discovering a Volume Offset Control Service instance. */
+struct bt_vocs_discover_param {
+	/**
+	 * @brief The start handle of the discovering.
+	 *
+	 * Typically the @p start_handle of a @ref bt_gatt_include.
+	 */
+	uint16_t start_handle;
+	/**
+	 * @brief The end handle of the discovering.
+	 *
+	 * Typically the @p end_handle of a @ref bt_gatt_include.
+	 */
+	uint16_t end_handle;
+};
+
+/**
+ * @brief Get a free service instance of Volume Offset Control Service from the pool.
+ *
+ * @return Volume Offset Control Service instance in case of success or NULL in case of error.
+ */
+struct bt_vocs *bt_vocs_free_instance_get(void);
+
+/**
+ * @brief Get the service declaration attribute.
+ *
+ * The first service attribute returned can be included in any other GATT service.
+ *
+ * @param vocs Volume Offset Control Service instance.
+ *
+ * @return Pointer to the attributes of the service.
+ */
+void *bt_vocs_svc_decl_get(struct bt_vocs *vocs);
+
+/**
+ * @brief Initialize the Volume Offset Control Service instance.
+ *
+ * @param vocs      Volume Offset Control Service instance.
+ * @param init      Volume Offset Control Service initialization structure.
+ *                  May be NULL to use default values.
+ *
+ * @return 0 if success, ERRNO on failure.
+ */
+int bt_vocs_init(struct bt_vocs *vocs, const struct bt_vocs_init_param *init);
+
+/**
+ * @brief Callback function for the offset state.
+ *
+ * Called when the value is read, or if the value is changed by either the server or client.
+ *
+ * @param conn        Connection to peer device, or NULL if local server read.
+ * @param inst        The instance pointer.
+ * @param err         Error value. 0 on success, GATT error or ERRNO on fail.
+ *                    For notifications, this will always be 0.
+ * @param offset      The offset value.
+ */
+typedef void (*bt_vocs_state_cb_t)(
+	struct bt_conn *conn, struct bt_vocs *inst, int err, int16_t offset);
+
+/**
+ * @brief Callback function for setting offset.
+ *
+ * @param conn        Connection to peer device, or NULL if local server write.
+ * @param inst        The instance pointer.
+ * @param err         Error value. 0 on success, GATT error on fail.
+ */
+typedef void (*bt_vocs_set_offset_cb_t)(struct bt_conn *conn, struct bt_vocs *inst, int err);
+
+/**
+ * @brief Callback function for the location.
+ *
+ * Called when the value is read, or if the value is changed by either the server or client.
+ *
+ * @param conn         Connection to peer device, or NULL if local server read.
+ * @param inst         The instance pointer.
+ * @param err          Error value. 0 on success, GATT error or ERRNO on fail.
+ *                     For notifications, this will always be 0.
```

Comment:
How do you differentiate between GATT errors and errno? Positive value is GATT, negative is -errno? If so, please document it clearly. Same for all other callbacks with a similar parameter.

Reply:
Your assumption is correct: 0 = success, > 0 GATT and < 0 ERRNO. Updated.